Remove export v1

jameshadfield opened the pull request:
Would anyone who enjoys diving into `argparse` specifics like to help out with this PR? My goal is to make `augur export` a synonym for `augur export v2` - the approach as implemented here nearly achieves this but fails for `augur export v2 <args>` because we have now set required arguments to the `augur export` parser itself so we get errors like:

```console
$ augur export v2 --tree X
augur export: error: the following arguments are required: --tree/-t
```

I presume a better approach is to set a default subparser?

victorlin commented:
> Would anyone who enjoys diving into `argparse` specifics like to help out with this PR? My goal is to make `augur export` a synonym for `augur export v2` - the approach as implemented here nearly achieves this but fails for `augur export v2 <args>` because we have now set required arguments to the `augur export` parser itself so we get errors like:
> 
> ```
> $ augur export v2 --tree X
> augur export: error: the following arguments are required: --tree/-t
> ```
> 
> I presume a better approach is to set a default subparser?

This is tricky. I don't think it's easy to share the same argument between two parsers (`export` and `export v2`). In your example, `augur export v2 --tree X` doesn't work because the `--tree` argument is always handled by the `export` parser so the `export v2` parser won't get any arguments. If you tweak it to work, I imagine that `augur export --tree X` will stop working because the argument would then be handled by the `export  v2` parser.

I think it'd be easier to simply have just one parser for `export`, and make `v2` an optional no-op argument. Something like this: ace1d81657120b989c9f973b5dec34fe99901d07 I haven't figured out how show the note you've added in the current implementation, but maybe it's possible or we could live without it.

tsibley commented:
What @joverlee521 said. Everything is using `augur export v2`, so we can just remove `augur export v1` and leave `augur export` and everything else as-is.

jameshadfield commented:
I think splitting `augur export` into `augur export {v1,v2}` wasn't the right move in hindsight and would like to remedy that. However given the 2 messages expressing disagreement with this view (and one emoji) I am going to reduce the scope of this PR in order to remove the deprecated `v1` subcommand. 👀  on the PR welcome!

jameshadfield commented on DEPRECATED.md:
```diff
@@ -26,8 +26,7 @@ We recognize the existing usage of this function, so it has been moved to
 
-*Deprecated in version 22.2.0 (July 2023). Planned for [removal](https://github.com/nextstrain/augur/issues/1266)
-January 2024 or after.*
+*Deprecated in version 22.2.0 (July 2023). Removed in version TKTK (TKTK 2025)*
```
[todo]

victorlin commented:
@jameshadfield I started this a while back but never got around to creating a PR. Feel free to cherry-pick any commits: db54927b...ec16b56e

tsibley commented:
> I think splitting `augur export` into `augur export {v1,v2}` wasn't the right move in hindsight and would like to remedy that.

Let's discuss this somewhere? I'm curious to hear more.

> I am going to reduce the scope of this PR in order to remove the deprecated `v1` subcommand.

:+1:

joverlee521 commented on docs/redirects.yaml:
```diff
@@ -82,7 +82,11 @@
-  to_url: /api/developer/augur.export_v1.html
+  to_url: /api/developer/augur.export_v2.html
+
+- type: page
+  from_url: /api/augur.export_v1.html
+  to_url: /api/developer/augur.export_v2.html
```
Looks like a duplicate entry as above, did you mean 

```suggestion
- type: page
  from_url: /api/developer/augur.export_v1.html
  to_url: /api/developer/augur.export_v2.html
```

jameshadfield replied:
Totally! I've also updated the two redirects to point to the migration guide (thanks @victorlin!)

joverlee521 reviewed (approved):
LGTM. I will note that we will need to [address the removal of augur export v1 in seasonal flu](https://github.com/nextstrain/seasonal-flu/issues/213) before/when this gets released. 

Also saying this out loud to confirm: We are keeping `augur validate export-v1` and the v1 JSON schemas to continue to support existing v1 datasets.

jameshadfield commented:
> [@victorlin] Feel free to cherry-pick any commits: https://github.com/nextstrain/augur/compare/db54927bd4f80168cc2334adce26273fc89772b0...ec16b56eefb3ac8df0382f39f4683d6a6d059987

> [@joverlee521] We are keeping augur validate export-v1 and the v1 JSON schemas to continue to support existing v1 datasets.

No - this was an oversight on my behalf. Victor's patch series removed this and I've now done the same in this PR. For completeness we've now:

* Removed `augur export v1`
* Removed `augur validate export-v1` (I guess I should add this to the changelog)
* Removed / updated relevant schemas. Victor's patch series also removed the parsing of old (v1-style) auspice-config JSONs within `augur export v2` but I've preserved this as I think there's still quite a few such JSONs in use. 
* Updated docs & redirects as appropriate

joverlee521 commented on augur/data/schema-export-v1-meta.json:
FYI, we still link to the v1 schemas in nextstrain.org and auspice ([GH search](https://github.com/search?q=org%3Anextstrain+schema-export-v1-meta.json+%28NOT+repo%3Anextstrain%2Faugur%29&type=code)). We should update those links to point to a specific tag instead of the default/master branch.
